### ISSUE TITLE
Null KML Feature for OnFeatureClickListener for All but LAST KML Added

### ISSUE THREAD

**Issue by johnknelsonintific** [googlemaps/android-maps-utils#466]
**Summary:** 

My Android application includes a Google Map which I want to display multiple KML files. I store them in assets and iterate through them. I then create a layer from each file, add it to map, and set an OnFeatureClickListener for each. The code is below:

```
         ...
         for (int i = 0; i< kmlList.length; i++) {
                InputStream is = assetManager.open("kml/"+kmlList[i]);
                KmlLayer layer = new KmlLayer(googleMap, is, getContext());
                layer.addLayerToMap();
                layer.setOnFeatureClickListener(new KmlLayer.OnFeatureClickListener() {
                    @Override
                    public void onFeatureClick(Feature feature) {
                        if(feature != null) {
                            Log.i("KmlClick", "Feature clicked: " + feature.getProperty("name"));
                        }
                    }
                });
            }
          ...
```

The issue is that the returned Feature for the on click method is null for all KML features *except* the last one added. I mostly want to know if this is a known issue or on the backlog to fix, or if there is a workaround. Otherwise I plan on creating Polylines from the polygons within my KML - just wanted to check the status first. Thanks!

**Steps to reproduce:** 

Create a layer, add to GoogleMap, and register a callback for that layer. Do this for multiple layers.

**Expected behavior:** 

Register an OnFeatureClickListener for each layer I added to the map which provides the feature that was clicked as an argument in the callback.

**Observed behavior:** 

The callback is triggered, but the Feature object is null for all layers *except* the last layer which was added.

**Device and Android version:** 

Samsung Galaxy S8 with Android 8.0.0 .

**Comment by ydrojen** [comment]
I have the same issue, onFeatureClick is triggered with null feature, but I only add one layer...

**Comment by ghopretz** [comment]
I have the same issue. Is there any solution?

**Comment by ghopretz** [comment]
problem solved, i use geojson file instead kml file.
i convert shapefile to geojson.

**Comment by jlafitte** [comment]
Just closed but it seems like there was no fix?  I'm getting this to.  I'll switch to geojson if I have to, but wondering if this is not an issue?

**Comment by johnknelsonintific** [comment]
> Just closed but it seems like there was no fix? I'm getting this too. I'll switch to geojson if I have to, but wondering if this is not an issue?

@jlafitte I suppose there was never a fix. I'm checking how I ended up resolving it, I ended up using one KML layer for all my polygons and handling _polygon_ clicks.

**Comment by jlafitte** [comment]
I have just one KML layer but it's all polylines. As far as I can tell Feature is always null, but the event fires fine.

**Comment by johnknelsonintific** [comment]
@jlafitte Ohhh right. Now I'm remembering, I only needed one string worth of information associated with each polygon, so I used `polygon.setTag(placemark.getProperty("exampleString"));//Store some information with the polyline` to store the data I needed to reference, rather than "features" which like you're seeing, are null for all but the last. Don't know if that helps, that's just my workaround.
